src/qseed.cpp

```diff
@@ -24,73 +30,104 @@ struct CommonConfig {
 
 };
 
-CommonConfig getCommonConfig() {
+CommonConfig getCommonConfig(std::string yamlContents) {
 
     CommonConfig commonConfig(DEFAULT_QSEED_SIZE, DEFAULT_QSEED_PERIOD);
-
-    const char* qryptToken = std::getenv("QRYPT_TOKEN");
-    if (!qryptToken) {
-        throw std::runtime_error("QRYPT_TOKEN environment variable is not set");
+    ryml::Tree tree = ryml::parse_in_place((char*)yamlContents.c_str());
+
+    const char* qryptTokenKey = "qrypt_token";
+    if (tree[qryptTokenKey].has_key() && tree[qryptTokenKey].has_val()) {
+        ryml::from_chars(tree[qryptTokenKey].val(), &commonConfig.qryptToken);
+    } 
+    else {
+        throw std::runtime_error(std::string(qryptTokenKey) + " is not set in configuration file.");
     }
-    commonConfig.qryptToken = qryptToken;
 
-    const char* qseedSizeAsStr = std::getenv("QSEED_SIZE");
-    if (qseedSizeAsStr) {
-        commonConfig.qseedSize = std::stoi(qseedSizeAsStr);
+    const char* qseedSizeKey = "size";
+    if (tree[qseedSizeKey].has_key() && tree[qseedSizeKey].has_val()) {
+        ryml::from_chars(tree[qseedSizeKey].val(), &commonConfig.qseedSize);
     }
     if (commonConfig.qseedSize == 0 || commonConfig.qseedSize > MAX_QSEED_SIZE) {
-        std::string errMsg = "QSEED_SIZE must be greater than 0 and less than or equal to " + std::to_string(MAX_QSEED_SIZE) + ".";
+        std::string errMsg = "Qseed size must be greater than 0 and less than or equal to " + std::to_string(MAX_QSEED_SIZE) + ".";
         throw std::runtime_error(errMsg);
     }
 
-    const char* qseedPeriodAsStr = std::getenv("QSEED_PERIOD");
-    if (qseedPeriodAsStr) {
-        commonConfig.qseedPeriod = std::stoi(qseedPeriodAsStr);
+    const char* qseedPeriodKey = "period";
+    if (tree[qseedPeriodKey].has_key() && tree[qseedPeriodKey].has_val()) {
+        ryml::from_chars(tree[qseedPeriodKey].val(), &commonConfig.qseedPeriod);
     }
     if (commonConfig.qseedPeriod == 0 || commonConfig.qseedPeriod > MAX_QSEED_PERIOD) {
-        std::string errMsg = "QSEED_PERIOD must be greater than 0 and less than or equal to " + std::to_string(MAX_QSEED_PERIOD) + ".";
+        std::string errMsg = "Qseed period must be greater than 0 and less than or equal to " + std::to_string(MAX_QSEED_PERIOD) + ".";
         throw std::runtime_error(errMsg);
     }
 
     return commonConfig;
 
 }
 
-CryptokiConfig getCryptokiConfig() {
+CryptokiConfig getCryptokiConfig(std::string yamlContents) {
 
     CryptokiConfig cryptokiConfig = {};
-
-    const char* libraryFile = std::getenv("CRYPTOKI_LIB");
-    if (!libraryFile) {
-        throw std::runtime_error("CRYPTOKI_LIB environment variable is not set");
+    ryml::Tree tree = ryml::parse_in_place((char*)yamlContents.c_str());
+
+    const char* libraryFileKey = "cryptoki_lib";
```

Comment:
Andrew - could optionally create a utility function for getting members from the yaml file given how simple the structure is.